truffle/src/main/java/org/jruby/truffle/runtime/loader/SourceCache.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2015 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+
+package org.jruby.truffle.runtime.loader;
+
+import com.oracle.truffle.api.source.Source;
+
+import java.io.IOException;
+import java.util.*;
+
+public class SourceCache {
```

Comment:
Mainly for the debugger. If I say I want to break on `foo.rb`, I need to know what files we have loaded that could fuzzily match that name.

What we have here isn't an overhead anyway as at the moment Truffle is also caching sources to some extent. They both probably need to get some eviction capability.